File: Backend/Travle.Model/Responses/PopularDestinationsReport.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// The "most popular destinations by period" report: the ranked rows plus the period they were
    /// computed over (echoed back so the PDF header and the on-screen table agree). Used by both the
    /// JSON preview endpoint and the PDF endpoint.
    /// </summary>
    public class PopularDestinationsReport
    {
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }

        /// <summary>Optional category the report was filtered to (null = all categories).</summary>
        public string? CategoryName { get; set; }

        public List<PopularDestinationRow> Rows { get; set; } = new();
    }
}

File: UI/travle_core/lib/src/models/popular_destinations_report.dart
import 'package:json_annotation/json_annotation.dart';

import 'popular_destination_row.dart';

part 'popular_destinations_report.g.dart';

/// The popular-destinations report (mirrors the backend `PopularDestinationsReport`):
/// the ranked rows plus the period and optional category the report was computed over.
@JsonSerializable()
class PopularDestinationsReport {
  PopularDestinationsReport({
    required this.rows,
    this.fromDate,
    this.toDate,
    this.categoryName,
  });

  final List<PopularDestinationRow> rows;
  final DateTime? fromDate;
  final DateTime? toDate;
  final String? categoryName;

  factory PopularDestinationsReport.fromJson(Map<String, dynamic> json) =>
      _$PopularDestinationsReportFromJson(json);

  Map<String, dynamic> toJson() => _$PopularDestinationsReportToJson(this);
}
